src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1139,12 +1141,12 @@
   [orig-report :- reports/report-schema
    timestamp :- pls/Timestamp
    update-latest-report? :- s/Bool]
-  (time! (:store-report metrics)
+  (time! (:store-report performance-metrics)
          (let [{:keys [puppet_version certname report_format configuration_version
                        start_time end_time resource_events transaction_uuid environment
-                       status noop]
-                :as report}         (normalize-report orig-report)
-                report-hash         (shash/report-identity-hash report)]
+                       status noop metrics logs] :as report} (normalize-report orig-report)
```

Comment:
I feel like this is kind of growing big enough you could just use a `select-keys` and `kitchensink.core/maptrans` to simplify this quite a bit!